### DISCUSSION TITLE
Anyone know the input/output format for these auto-encoder onnx?

### DISCUSSION THREAD

**Discussion by elephantpanda** [microsoft/onnxruntime#14217] [General]
I am running [these autoencoder onnx using Onnxruntime in c#](https://huggingface.co/bes-dev/stable-diffusion-v1-4-onnx/tree/main):

I am trying to encode/decode an image of a cat just using the vae-encoder and decoder. 

I am getting a blurry outcome of the reconstituted cat. 

I wonder if anyone knows the correct format for the input and output tensors?

I began by encoding my image as RGB with float32 values in the range [-1,1]
And the output of the vae-decoder I was running through a sigmoid then multiplying by 255 to get pixel values.

I have a feeling this is not correct. Does anyone know the correct format for the input/output tensors?

I could even be in the wrong colour-space for example maybe it wants HSL?

**Comment by xadupre** [comment]
Did you try in the range [0, 1]?
